src/accordion/accordion.ts

```diff
@@ -321,6 +364,26 @@ export class NgbAccordion implements AfterContentChecked {
   private _updateActiveIds() {
     this.activeIds = this.panels.filter(panel => panel.isOpen && !panel.disabled).map(panel => panel.id);
   }
+
+  private _runTransition(panel: NgbPanel) {
+    panel.transitionRunning = true;
+    this._changeDetector.detectChanges();
+    this._ngZone.onStable.pipe(take(1)).subscribe(() => {
```

Comment:
1. I don't think we need this `onStable` here at all. Everything is fine in my tests if we don't do it + you can remove all double `fixture.detectChanges()` from the tests.

Adding those `fixture.detectChanges()` feels like something is wrong at the first place :)

2. Also we need to run `this._changeDetector.detectChanges()` only once, not for each panel in a loop !!! (in case of `_closeOthers`)

Reply:
> + you can remove all double fixture.detectChanges() from the tests

👍

Reply:
1. It seems so. I remember I've added the onStable because of issues, but I think it was more due to `this._changeDetector.detectChanges()` I forgot to add.
2. Not sure how to handle it. The purpose of detectChanges is to ensure that `panel.transitionRunning = true;` is well reflected in the dom, just before the runTransition. So if I do it after the loop, it's too late. Need more time to think about it :)

Reply:
Run it before the loop IF there are any open tabs, no?

Reply:
:smile: I achieved almost the same code before reading your comment, the variable name is the only difference ! In this case, there are still 2 changeDetections if it's run from _changeOpenState (one for closeOthers, one for the active panel).

We can also improve that by:

- Changing the isOpen state of the panel to the desired state,
- Setting runningTranstion to true on the panel,
- run _runTransitions to perform the changeDetection (once) then run the pending transitions in a row.

This will be optimum, but I wonder if the additional complexity worth it.

What do you think ?

Reply:
ok, after a discussion with @maxokorokov, I'll try the optimization